tests/utils.go

```diff
@@ -0,0 +1,172 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2017 Red Hat, Inc.
+ *
+ */
+
+package tests
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	k8sv1 "k8s.io/api/core/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	
+	"k8s.io/apimachinery/pkg/api/errors"
+
+	"kubevirt.io/kubevirt/pkg/kubecli"
+	"kubevirt.io/kubevirt/pkg/log"
+
+)
+
+var KubeVirtVersionTag = "latest"
+var KubeVirtRepoPrefix = "kubevirt"
+var KubeVirtKubectlPath = ""
+var KubeVirtOcPath = ""
+var KubeVirtInstallNamespace = "kube-system"
+
+func init() {
+	flag.StringVar(&KubeVirtVersionTag, "tag", "latest", "Set the image tag or digest to use")
+	flag.StringVar(&KubeVirtRepoPrefix, "prefix", "kubevirt", "Set the repository prefix for all images")
+	flag.StringVar(&KubeVirtKubectlPath, "kubectl-path", "", "Set path to kubectl binary")
+	flag.StringVar(&KubeVirtOcPath, "oc-path", "", "Set path to oc binary")
+	flag.StringVar(&KubeVirtInstallNamespace, "installed-namespace", "kube-system", "Set the namespace KubeVirt is installed in")
+}
+
+const (
+	// tests.NamespaceTestDefault is the default namespace, to test non-infrastructure related KubeVirt objects.
+	NamespaceTestDefault = "cdi-test-default"
+	// NamespaceTestAlternative is used to test controller-namespace independency.
+	NamespaceTestAlternative = "cdi-test-alternative"
+)
+
+const (
+	SecretLabel = "kubevirt.io/secret"
+)
+
+var testNamespaces = []string{NamespaceTestDefault, NamespaceTestAlternative}
+
+// KubevirtFailHandler call ginkgo.Fail with printing the additional information
+func KubevirtFailHandler(message string, callerSkip ...int) {
+	if len(callerSkip) > 0 {
+		callerSkip[0]++
+	}
+	Fail(message, callerSkip...)
+}
+
+func BeforeTestSuitSetup() {
+
+	log.InitializeLogging("tests")
+	log.Log.SetIOWriter(GinkgoWriter)
+
+	createNamespaces()
+}
+
+func AfterTestSuitCleanup() {
+	// Make sure that the namespaces exist, to not have to check in the cleanup code for existing namespaces
+	createNamespaces()
+	cleanNamespaces()
+	removeNamespaces()
+}
+
+func BeforeTestCleanup() {
+	cleanNamespaces()
+}
+
+func cleanNamespaces() {
+}
+
+func createNamespaces() {
+	virtCli, err := kubecli.GetKubevirtClient()
+	PanicOnError(err)
+
+	// Create a Test Namespaces
+	for _, namespace := range testNamespaces {
+		ns := &k8sv1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: namespace,
+			},
+		}
+		_, err = virtCli.CoreV1().Namespaces().Create(ns)
+		if !errors.IsAlreadyExists(err) {
+			PanicOnError(err)
+		}
+	}
+}
+
+func removeNamespaces() {
+	virtCli, err := kubecli.GetKubevirtClient()
```

Comment:
use the Kubeernetes Client for this, not the KubeVirt wrapper around the kubernetes client. We don't need kubecli, we can use kubevirt's client-go directly

Reply:
Yes, see other comment, replaced with kubernetes one.